src/main/java/org/blacksmith/entitytypes/OnScreenEntity.java

```diff
@@ -0,0 +1,86 @@
+package org.blacksmith.entitytypes;
+
+import java.awt.*;
+
+public abstract class OnScreenEntity extends Entity {
+    public int x, y;
+
+    Image image;
+
+    int width, height;
+
+    int velocityX;
+    int velocityY;
+
+    public int getX() {
+        return x;
+    }
+
+    public void setX(int x) {
+        this.x = x;
+    }
+
+    public int getY() {
+        return y;
+    }
+
+    public void setY(int y) {
+        this.y = y;
+    }
+
+    public int getWidth() {
+        return width;
+    }
+
+    public void setWidth(int width) {
+        this.width = width;
+    }
+
+    public int getHeight() {
+        return height;
+    }
+
+    public void setHeight(int height) {
+        this.height = height;
+    }
+
+    public int getVelocityY() {
+        return velocityY;
+    }
+
+    public void setVelocityY(int velocityY) {
+        this.velocityY = velocityY;
+    }
+
+    public int getVelocityX() {
+        return velocityX;
+    }
+
+    public void setVelocityX(int velocityX) {
+        this.velocityX = velocityX;
+    }
+
+    public int scaleX(float virtualWidth, int screenWidth) {
```

Comment:
what's this scale stuff?  you want to be able to resize the screen, or work on mobile devices with different screen sizes.  eventually, we'll use this.